app/api/web/fetch_file.coffee

```diff
@@ -0,0 +1,96 @@
+#include all the required dependencies
+util = require('../../common/util')
+appconfig = require('../../config/appconfig').appconfig
+fs = require('fs')
+request = require('request')
+csrf = require('csurf')
+async = require('async')
+prettysize = require('prettysize')
+path = require('path')
+tmpDir = appconfig.getTemporaryDirectory()
+fileLimits = appconfig.getFileLimits()
+filestores = {}
+mimeRegex = /.(avi|wmv|flv|mpg|3gp|mkv|mp4|mpeg|mpeg-1|mpeg-2|mpeg-3|mpeg-4|mp3|wav|xlsx?|zip|7z|docx?|pptx?|pdf)$/i
+
+exports.init = (app) ->
+	
+	app.post '/fetch/', csrf({ cookie:true }), (req,res) ->
+		#check request came
+		console.log "Request recieved"
+		#set file name from req.body
+		fileName = path.basename(req.body.url)
+		#Remove unnecessary characters and spaces in the file
+		fileName = fileName.replace(/@[^0-9a-z\.]+@i/g, "-")
+		fileName = fileName.replace(/\ +/g, "-")
+		#give file a unique id ending with the file name , provided it has one.
+		fileName = "#{util.uniqueId()}-#{fileName}"
+		reqURL = [req.body.url]
+		#First GET request for Headers
+		responseObject = {}
+		#Check user allowed Mime Types
+		if req.body.allowedMimeTypes != "*/*"
+			fileRegex = req.body.allowedMimeTypes
+		else
+			fileRegex = mimeRegex
+        
+		fetchFile = (reqURL, callback) -> 
+			dataSize = 0
+			request(reqURL,{method : 'HEAD'}, (err, res) ->
+				if err
+					console.error err
+					responseObject =
+						error: true
+						message: err.message
+					callback(null)
+				else
+					console.log "got headers"
```

Comment:
Removed all console.logs since currently don't have a way to find the context of the log.